content/blog/2018-11-11-colocr.md

```diff
@@ -0,0 +1,106 @@
+
+## Developing a shiny app in an R package
+
+At first, I thought of adding a shiny app as an extra feature for users who might need it. I quickly realized that for this use case, selecting regions of interest might require trial-and-error on the side of the user to arrive at suitable parameters for the selection. Having to develop the app forced me to think about the organizing the functions, their input and output in a certain way. Namely, mapping the functions to the steps of the analysis that the user has to think about and making visual output a priority. Finally, testing the app exposed errors in the intermediates which might have been buried in an otherwise seemingly fine [R](https://cran.r-project.org) objects. Overall, I learned a thing or two about writing shiny apps to go with [R](https://cran.r-project.org) packages when possible even if it won't make the cut in the release.
+
+* I found it useful to work on the app as early as possible and improving it side by side with the core functionality. As explained earlier, this helped me to think about the inputs and the output from a user's perspective.
+*  I learnt to isolate the source code for the app from that of the package functions as much as possible. One may host the app code in a separate repository and including it in the project as a **git submodule**. Some might find this necessary, but I found it easier this way to keep tracking of the changes in the app and testing it on Travis.
+
+## Submitting R packages for reviewing on ROpenSci
+
+`colocr` was my second submission on [ROpenSci](https://ropensci.org). In both cases, I had positive experiences. So I would like to thank the [ROpenSci](https://ropensci.org) editor, the reviewers ([@haozhu233](https://github.com/haozhu233), [@seaaan](https://github.com/seaaan)) and [@jeroen](https://github.com/jeroen) who helped us to resolve issues related to running the package on windows. At a risk of an understatement, here is three benefits I got from submitting my packages to [ROpenSci](https://ropensci.org) which others might find encouraging doing the same.
+
+* Reviewing the source could at an early stage help to spot mistakes and introducing improvement to the code that might escape the authors themselves.
+* Often the editor and the reviewers are experts in [R](https://cran.r-project.org) and the subject matter, so they can offer ideas and suggestion to improve the design of the package. Adding to being an early test users who provided useful insights.
+* Writing and distributing code might not be everyone's strongest skill. And certain aspects of it might be challenging. The [ROpenSci](https://ropensci.org) team provide support during the process I found encouraging.
+
+## Things to improve
```

Comment:
It's great that you've identified things to improve. Perhaps you could open them as issues in the repo and link to them - might draw people to the repo who might be interested to help.